services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/config/JdbcProperties.java

```diff
@@ -168,14 +176,40 @@ public static JDBCClient dataSource(final Vertx vertx, final JdbcProperties data
 
         log.info("Creating new SQL client: {} - table: {}", config, dataSourceProperties.getTableName());
 
-        // put password after logging
-
-        config
-            .put("password", dataSourceProperties.getPassword());
-
         // create new client
 
-        return JDBCClient.create(vertx, config);
+        final int minSize = config.getInteger(minSizeLabel);
+        final int maxSize = config.getInteger(maxSizeLabel);
+        final int initSize = config.getInteger(initSizeLabel);
+        final Duration connectionTimeout = Duration.ofSeconds(30);
+        final Duration idleTimeout = Duration.ofSeconds(config.getInteger(maxIdleLabel));
+        final NamePrincipal username = Optional
+                .ofNullable(dataSourceProperties.getUsername())
+                .map(NamePrincipal::new)
+                .orElse(null);
+        final SimplePassword password = Optional
+                .ofNullable(dataSourceProperties.getPassword())
+                .map(SimplePassword::new)
+                .orElse(null);
+
+        final AgroalDataSourceConfigurationSupplier configuration = new AgroalDataSourceConfigurationSupplier()
+                .metricsEnabled(false)
+                .dataSourceImplementation(AgroalDataSourceConfiguration.DataSourceImplementation.AGROAL)
+                .connectionPoolConfiguration(poolConfig -> poolConfig
+                        .minSize(minSize)
+                        .maxSize(maxSize)
+                        .initialSize(initSize)
+                        .acquisitionTimeout(connectionTimeout)
+                        .validationTimeout(connectionTimeout)
+                        .leakTimeout(idleTimeout)
+                        .reapTimeout(idleTimeout)
```

Comment:
Do you see a benefit in setting leakTimeout = reapTimeout and acquisitionTimeout = validationTimeout? Also I am wondering, do you know if there is a default validation method or would we have to provide one for the validation to work?

Reply:
After looking into these settings again and more closely I think current ones are not very optimal and here's what I'm thinking would work better here;

- acquisitionTimeout should be the configurable connection_timeout value (i.e no matter why acquiring a connection takes a long time it will timeout eventually)
- validationTimeout is possibly best keep a constant like 5 seconds or alike simply to execute background validation at certain interval
- leakTimeout could be also a constant around 1 minute to make it possible to monitor it no connections are leaked (which should not happen anyway)
- reapTimeout is the current idle_timeout
- connectionValidator I need to check how this needs to be set

Reply:
After thinking about this further I think we should make all of these settings configurable with a default, as I see no real downside to it. Or can you think of any? As for the default on the validationTimeout, I am not sure how computationally expensive the connection validation is but I think validating every connection every 5 seconds could be a bit to expensive. Do you have more insight by now into how a connectionValidator can be set and if there is a default one configured?

Reply:
Ok, I try to make these configurable changes for validationTimeout and leakTimeout as soon as possible. The validator was something I needed to fix indeed but it is part of this PR already and it uses a default validator now - it was noop-validation by default.

Reply:
I just merged the last changes requested and follow that this PR still compiles;
- Both validationTime and leakTime are configurable
- Defaults are validationTime=30s and leakTime=60s - both can be disable with value zero